graphql_compiler/compiler/emit_sql.py

```diff
@@ -693,16 +706,34 @@ def __init__(self, sql_schema_info, ir):
         self._outputs = []  # sqlalchemy Columns labelled correctly for output
         self._filters = []  # sqlalchemy Expressions to be used in the where clause
 
-        self._current_fold = None  # SQLFoldObject to collect fold info and guide output query
-        self._fold_vertex_location = None  # location in the IR tree where the fold starts
-
         self._alias_generator = UniqueAliasGenerator()  # generates aliases for the fold subqueries
 
     def _relocate(self, new_location):
-        """Move to a different location in the query, updating the _alias."""
+        """Move to a different location in the query, updating the _current_alias."""
         self._current_location = new_location
-        if (self._current_location.query_path, None) in self._aliases:
-            self._current_alias = self._aliases[(self._current_location.query_path, None)]
+        # Create appropriate alias key based on whether or not relocation is taking place in a fold
+        # or not. Note: during end_fold relocation to outside of the fold (i.e. to a Location rather
+        # than a FoldScopeLocation) occurs before self._current_fold is set to None. Therefore, it
+        # should be checked that the current fold is not None AND that the new location is a
+        # FoldScopeLocation.
```

Comment:
Is this function ever called with a `FoldScopeLocation` and  `_current_fold` being `None`? Seems like `fold`, `unfold` and `traverse (inside fold)` all have `_current_fold` set when calling it.

If no, I'd remove `_current_fold` logic from this function.